### ISSUE TITLE
Defining `dataset` in a relation results in not defining methods like `by_pk`

### ISSUE THREAD

**Issue by backus** [rom-rb/rom-sql#210]
Example:

```ruby
require 'rom'
require 'rom/sql'

ROM::VERSION # => "3.2.0"
ROM::SQL::VERSION # => "1.2.2"

rom1 =
  ROM.container(:sql, 'sqlite::memory') do |conf|
    conf.default.create_table(:users) do
      primary_key :id
      column :name, String, null: false
      column :email, String, null: false
    end

    conf.relation(:users) do
      schema(infer: true)
    end
  end

rom2 =
  ROM.container(:sql, 'sqlite::memory') do |conf|
    conf.default.create_table(:users) do
      primary_key :id
      column :name, String, null: false
      column :email, String, null: false
    end

    conf.relation(:users) do
      schema(infer: true)

      dataset do
        select(:id, :name)
      end
    end
  end


rom1.relations.users.insert(id: 1, name: 'Alice', email: 'alice@example.cmo')
rom2.relations.users.insert(id: 1, name: 'Alice', email: 'alice@example.cmo')

rom1.relations.users.first # => {:id=>1, :name=>"Alice", :email=>"alice@example.cmo"}
rom2.relations.users.first # => {:id=>1, :name=>"Alice"}

rom1.relations.users.by_pk(1) # => #<ROM::Relation[Users] dataset=#<Sequel::SQLite::Dataset: "SELECT `id`, `name`, `email` FROM `users` WHERE (`id` = 1) ORDER BY `users`.`id`">>
rom2.relations.users.by_pk(1)
# ~> /Users/johnbackus/.gem/ruby/2.4.0/gems/rom-3.2.0/lib/rom/plugins/relation/registry_reader.rb:28:in `block in method_missing': undefined method `by_pk' for #<#<Class:0x007f873d0c17c0>:0x007f873d10f5b0> (NoMethodError)
# ~>  from /Users/johnbackus/.gem/ruby/2.4.0/gems/rom-3.2.0/lib/rom/registry.rb:33:in `block in fetch'
# ~>  from /Users/johnbackus/.gem/ruby/2.4.0/gems/rom-3.2.0/lib/rom/registry.rb:32:in `fetch'
# ~>  from /Users/johnbackus/.gem/ruby/2.4.0/gems/rom-3.2.0/lib/rom/registry.rb:32:in `fetch'
# ~>  from /Users/johnbackus/.gem/ruby/2.4.0/gems/rom-3.2.0/lib/rom/plugins/relation/registry_reader.rb:28:in `method_missing'
# ~>  from -:45:in `<main>'
```

**Comment by solnic** [comment]
Thanks for reporting this. I'll try to fix it in `1.x` line and port it to master. Worst case I'll only fix it in master and it'll be in rom-sql 2.0.

**Comment by solnic** [comment]
This was fixed in master, although I didn't add a spec for it yet (the fix was a nice side-effect of a recent refactoring). I'll add a spec and close this one. Fixing it in 1.x line will be too hard, so I hope you can live with that and you'll be all good once you upgrade to 4.0.
